components/nimbus/src/lib.rs

```diff
@@ -145,7 +145,12 @@ impl NimbusClient {
         state: &mut MutexGuard<InternalMutableState>,
     ) -> Result<()> {
         self.update_ta_install_dates(db, writer, state)?;
-        self.event_store.lock().unwrap().read_from_db(db)?;
+        let mut event_store = self.event_store.lock().unwrap();
+        event_store.read_from_db(db)?;
+
+        // Now that the EventStore has been created, place it in its current state on the targeting_attributes.
+        let mut state = self.mutable_state.lock().unwrap();
+        state.targeting_attributes.event_store = Some(event_store.clone());
```

Comment:
Going to really quickly comment here to share one thing I'm still not sure of -- I saw two options when it comes to putting the event store on the targeting attributes. 
1. As soon as it exists and every time it is updated, clone it and put it on the targeting attributes
2. When we go to apply experiments or other methods that would use the targeting attributes, set the event store there, then after it has been used set it to `None`. 

This PR is currently using option 1, but I'm not sure whether that's the best choice. Thoughts would be appreciated!

Reply:
I think we really want to grab a snapshot of the event store when applying experiments, so I would say we just need to refresh the targeting_attributes copy right before it gets consumed as part of applying the updates. If If you do it here, there may be events that occur between now and the app calling `apply_updates` that would have influenced the targeting evaluation. I would argue that this should happen in `apply_updates` rather than here, for that reason.  

My thinking is that any time there is a query, we would capture a "snapshot" (.clone()) of the event_store and release the in-memory one so that events could continue to be recorded. This is exactly what you have done, but I think `initialize` is too far from where the query is being executed.

Reply:
Ah that's a good call -- yep I'll make that adjustment

Reply:
I've made that adjustment -- would you mind giving your thoughts on the new code?

Reply:
I'm planning on looking at this again this morning, sorry for any delay